pkg/resource/deploy/deployment_executor.go

```diff
@@ -233,6 +250,16 @@ func (ex *deploymentExecutor) Execute(callerCtx context.Context, opts Options, p
 				}
 
 				if event.Event == nil {
+					if workerPool != nil {
+						// Wait for all currently pending and executing workers to
+						// complete. Then check if any worker had an error.
+						poolErr := workerPool.Wait(true)
+						if poolErr != nil {
+							cancel()
```

Comment:
I'm curious about the behaviour in case errors happen here.  It looks to me that this might unintentionally turn on `--continue-on-error` behaviour for regular deploys (`pulumi up` only) as well.  We're waiting until all the events are done to cancel the deployment, meaning we'll keep processing events even after an error happened.  It's probably inevitable that that's happening to a degree with parallelism, but curious if that's what you intended here?

Edit: ah I see the worker pool cancels itself when there's any errors, so this is probably fine.